docs/user-guide/migrating-to-7.0.0.md

```diff
@@ -0,0 +1,236 @@
+
+ESLint v7.0.0 is a major release of ESLint. We have made a few breaking changes in this release. This guide is intended to walk you through the breaking changes.
+
+The lists below are ordered roughly by the number of users each change is expected to affect, where the first items are expected to affect the most users.
+
+## Table of Content
+
+### Breaking changes for users
+
+- [Node.js 8 is no longer supported](#drop-node-8)
+- [Lint the files that match to `overrides[].files` by default](#additional-lint-targets)
+- [The base path of `overrides` and `ignorePatterns` is changed if the config file is given by the `--config`/`--ignore-path` options](#base-path-change)
+- [The place where ESLint loads plugins from is changed](#plugin-loading-change)
+- [Runtime deprecation warnings on `~/.eslintrc.*` config files](#runtime-deprecation-on-personal-config-files)
+- [Default ignore patterns are changed](#default-ignore-patterns)
+- [Description in directive comments](#description-in-directive-comments)
+- [Node.js/CommonJS rules are deprecated](#deprecate-node-rules)
+- [Several rules get strict](#rules-strict)
+- [`eslint:recommended` is updated](#eslint-recommended)
+
+### Breaking changes for plugin developers
+
+- [Node.js 8 is no longer supported](#drop-node-8)
+- [Lint the files that match to `overrides[].files` by default](#additional-lint-targets)
+- [The place where ESLint loads plugins from is changed](#plugin-loading-change)
+- [`RuleTester` class get strict](#rule-tester-strict)
+
+### Breaking changes for integration developers
+
+- [Node.js 8 is no longer supported](#drop-node-8)
+- [The place where ESLint loads plugins from is changed](#plugin-loading-change)
+- [`CLIEngine` class has been deprecated](#deprecate-cliengine)
+
+---
+
+## <a name="drop-node-8"></a> Node.js 8 is no longer supported
+
+Node.js 8 has been at EOL since December 2019. Therefore, we have decided to drop support for it in ESLint 7. We now support the following versions of Node.js:
+
+- Node.js 10 (`10.12.0` and above)
+- Anything above Node.js 12
+
+**To address:** Make sure you upgrade to at least Node.js `10.12.0` when using ESLint 7. Mind the Node.js version of your editor when you use ESLint 7 via editor integrations. If you are unable to upgrade, we recommend continuing to use ESLint 6 until you are able to upgrade Node.js.
+
+**Related issue(s):** [RFC44](https://github.com/eslint/rfcs/blob/master/designs/2019-drop-node8/README.md), [#12700](https://github.com/eslint/eslint/pull/12700)
+
+## <a name="additional-lint-targets"></a> Lint the files that match to `overrides[].files` by default
+
+Previously, ESLint lints only `*.js` files by default if directories are present such as below.
+
+```
+$ eslint src
+```
+
+In the situation, ESLint 7 lints the files that match to `overrides[].files` along with `*.js` files. (But it excepts patterns that end with `*` in order to avoid linting unlimitedly.) For example, if the following config file is present,
```

Comment:
There’s a mix of “ESLint 7” and “v7.0.0” throughout this file. I’d suggest using “v7.0.0” everywhere to be as clear as possible.